### ISSUE TITLE
crash bug on Debian

### ISSUE THREAD

**Issue by ioerror** [ricochet-im/ricochet#276]
I woke up to my Ricochet (1.1.1 - rev ...9f6710549f7fe051e) crashed:
```
qrc:/ui/MainWindow.qml:89:17: QML Loader: Binding loop detected for property "active"
qrc:/ui/ContactRequestDialog.qml:21: TypeError: Cannot read property 'hostname' of null

qrc:/ui/ContactRequestDialog.qml:21: TypeError: Cannot read property 'hostname' of null
QCoreApplication::postEvent: Unexpected null receiver
=================================================================
==12391==ERROR: AddressSanitizer: heap-use-after-free on address 0x603001d648c0 at pc 0x7fcb2aac0fa3 bp 0x7ffc72a72e00 sp 0x7ffc72a72df8
READ of size 8 at 0x603001d648c0 thread T0
ASAN:SIGSEGV
==12391==AddressSanitizer: while reporting a bug found another one.Ignoring.
```

I have no tor log (tor was launched by Ricochet) - so i have no idea what was happening with Tor at the time of the crash.

**Comment by qbi** [comment]
I had the same today with version 1.1.0.

**Comment by armadev** [comment]
Here's the traceback I sent special on 25 Sep (he thinks it is the same bug):
```
=================================================================
==6582==ERROR: AddressSanitizer: heap-use-after-free on address 0x60300087d070 at pc 0x7fdd02730063 bp 0x7ffe0b8119d0 sp 0x7ffe0b8119c8
READ of size 8 at 0x60300087d070 thread T0
    #0 0x7fdd02730062 in Protocol::Connection::isConnected() const (/home/arma/old/torgit/ricochet/ricochet+0xb2062)
    #1 0x7fdd0273010a in Protocol::Connection::close() (/home/arma/old/torgit/ricochet/ricochet+0xb210a)
    #2 0x7fdd0273721c in Protocol::OutboundConnectorPrivate::abort() (/home/arma/old/torgit/ricochet/ricochet+0xb921c)
    #3 0x7fdd02737349 in Protocol::OutboundConnector::abort() (/home/arma/old/torgit/ricochet/ricochet+0xb9349)
    #4 0x7fdd026e9726 in ContactUser::updateOutgoingSocket() (/home/arma/old/torgit/ricochet/ricochet+0x6b726)
    #5 0x7fdd026e9ef9 in ContactUser::updateStatus() (/home/arma/old/torgit/ricochet/ricochet+0x6bef9)
    #6 0x7fdd026eb8ec in ContactUser::onConnected() (/home/arma/old/torgit/ricochet/ricochet+0x6d8ec)
    #7 0x7fdd0276d3a7 in ContactUser::qt_static_metacall(QObject*, QMetaObject::Call, int, void**) (/home/arma/old/torgit/ricochet/ricochet+0xef3a7)
    #8 0x7fdcfe875ed9 in QObject::event(QEvent*) (/usr/lib/x86_64-linux-gnu/libQt5Core.so.5+0x309ed9)
    #9 0x7fdcff87a12b in QApplicationPrivate::notify_helper(QObject*, QEvent*) (/usr/lib/x86_64-linux-gnu/libQt5Widgets.so.5+0x13f12b)
    #10 0x7fdcff87f5ff in QApplication::notify(QObject*, QEvent*) (/usr/lib/x86_64-linux-gnu/libQt5Widgets.so.5+0x1445ff)
    #11 0x7fdcfe845dca in QCoreApplication::notifyInternal(QObject*, QEvent*) (/usr/lib/x86_64-linux-gnu/libQt5Core.so.5+0x2d9dca)
    #12 0x7fdcfe847dc2 in QCoreApplicationPrivate::sendPostedEvents(QObject*, int, QThreadData*) (/usr/lib/x86_64-linux-gnu/libQt5Core.so.5+0x2dbdc2)
    #13 0x7fdcfe89dda2 (/usr/lib/x86_64-linux-gnu/libQt5Core.so.5+0x331da2)
    #14 0x7fdcfc962c5c in g_main_context_dispatch (/lib/x86_64-linux-gnu/libglib-2.0.so.0+0x49c5c)
    #15 0x7fdcfc962f47 (/lib/x86_64-linux-gnu/libglib-2.0.so.0+0x49f47)
    #16 0x7fdcfc962ffb in g_main_context_iteration (/lib/x86_64-linux-gnu/libglib-2.0.so.0+0x49ffb)
    #17 0x7fdcfe89e1b6 in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) (/usr/lib/x86_64-linux-gnu/libQt5Core.so.5+0x3321b6)
    #18 0x7fdcfe843511 in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) (/usr/lib/x86_64-linux-gnu/libQt5Core.so.5+0x2d7511)
    #19 0x7fdcfe84b15b in QCoreApplication::exec() (/usr/lib/x86_64-linux-gnu/libQt5Core.so.5+0x2df15b)
    #20 0x7fdd026a8efb in main (/home/arma/old/torgit/ricochet/ricochet+0x2aefb)
    #21 0x7fdcfd50bb44 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x21b44)
    #22 0x7fdd026ac316 (/home/arma/old/torgit/ricochet/ricochet+0x2e316)

0x60300087d070 is located 16 bytes inside of 24-byte region [0x60300087d060,0x60300087d078)
freed by thread T0 here:
    #0 0x7fdd015d9477 in operator delete(void*) (/usr/lib/x86_64-linux-gnu/libasan.so.1+0x55477)
    #1 0x7fdcfe875f17 in QObject::event(QEvent*) (/usr/lib/x86_64-linux-gnu/libQt5Core.so.5+0x309f17)
    #2 0x7fdcff87a12b in QApplicationPrivate::notify_helper(QObject*, QEvent*) (/usr/lib/x86_64-linux-gnu/libQt5Widgets.so.5+0x13f12b)

previously allocated by thread T0 here:
    #0 0x7fdd015d8fff in operator new(unsigned long) (/usr/lib/x86_64-linux-gnu/libasan.so.1+0x54fff)
    #1 0x7fdd02738f09 in Protocol::OutboundConnectorPrivate::onConnected() (/home/arma/old/torgit/ricochet/ricochet+0xbaf09)
    #2 0x7fdd0278e257 (/home/arma/old/torgit/ricochet/ricochet+0x110257)

SUMMARY: AddressSanitizer: heap-use-after-free ??:0 Protocol::Connection::isConnected() const
Shadow bytes around the buggy address:
  0x0c06801079b0: fa fa fa fa fa fa fa fa fd fd fd fa fa fa fd fd
  0x0c06801079c0: fd fd fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0c06801079d0: fa fa fa fa fa fa fa fa fa fa fa fa fd fd fd fd
  0x0c06801079e0: fa fa fd fd fd fd fa fa fd fd fd fd fa fa fa fa
  0x0c06801079f0: fa fa fa fa fa fa fa fa fa fa fd fd fd fd fa fa
=>0x0c0680107a00: fd fd fd fd fa fa fd fd fd fd fa fa fd fd[fd]fa
  0x0c0680107a10: fa fa fd fd fd fd fa fa fd fd fd fa fa fa fd fd
  0x0c0680107a20: fd fd fa fa fa fa fa fa fa fa fd fd fd fd fa fa
  0x0c0680107a30: fd fd fd fd fa fa fd fd fd fa fa fa fd fd fd fd
  0x0c0680107a40: fa fa fd fd fd fd fa fa fd fd fd fd fa fa fd fd
  0x0c0680107a50: fd fd fa fa fd fd fd fd fa fa fd fd fd fa fa fa
Shadow byte legend (one shadow byte represents 8 application bytes):
  Addressable:           00
  Partially addressable: 01 02 03 04 05 06 07
  Heap left redzone:       fa
  Heap right redzone:      fb
  Freed heap region:       fd
  Stack left redzone:      f1
  Stack mid redzone:       f2
  Stack right redzone:     f3
  Stack partial redzone:   f4
  Stack after return:      f5
  Stack use after scope:   f8
  Global redzone:          f9
  Global init order:       f6
  Poisoned by user:        f7
  Contiguous container OOB:fc
  ASan internal:           fe
==6582==ABORTING

[3]+  Exit 1                  ./ricochet  (wd: ~/old/torgit/ricochet)
```

**Comment by armadev** [comment]
(For context my traceback above was from git d29bb2b45ea.)

**Comment by special** [comment]
Resolved in 1.1.2
